hoot-core/src/main/cpp/hoot/core/conflate/highway/Roundabout.cpp

```diff
@@ -340,24 +349,87 @@ void Roundabout::replaceRoundabout(boost::shared_ptr<OsmMap> pMap)
     }
 
     // All our nodes should be there, now lets add the way back
-    WayPtr pNewWay(new Way(*_roundaboutWay));
+    WayPtr pRoundabout(new Way(*_roundaboutWay));
 
     // Make sure our nodes are set correctly
     std::vector<long> nodeIds;
     for (size_t i = 0; i < wayNodes.size(); i++)
       nodeIds.push_back(wayNodes[i]->getId());
-    pNewWay->setNodes(nodeIds);
-    pMap->addWay(pNewWay);
-  }
+    pRoundabout->setNodes(nodeIds);
+    pMap->addWay(pRoundabout);
+
```

Comment:
From here down is where the extra connecting is done.  All ways that were disconnected are looked at.  If they've been merged into another way and don't exist anymore the algorithm then finds the endpoint of the old way that used to connect to the roundabout and finds all ways that share that point.  There should only be two, one is the temporary way connecting it to the centerpoint and the other is the way that needs to be connected back up.  It is then connected back up.